exporters/otlp/otlp_test.go

```diff
@@ -283,46 +283,109 @@ func TestNewExportPipeline(t *testing.T) {
-	assert.Equal(t, 0, driverTraces.metricsExported)
-	assert.Equal(t, 0, driverMetrics.tracesExported)
-	assert.Equal(t, 1, driverMetrics.metricsExported)
-
-	assert.NoError(t, driver.Stop(ctx))
-	assert.Equal(t, 1, driverTraces.started)
-	assert.Equal(t, 1, driverMetrics.started)
-	assert.Equal(t, 1, driverTraces.stopped)
-	assert.Equal(t, 1, driverMetrics.stopped)
-	assert.Equal(t, 1, driverTraces.tracesExported)
-	assert.Equal(t, 0, driverTraces.metricsExported)
-	assert.Equal(t, 0, driverMetrics.tracesExported)
-	assert.Equal(t, 1, driverMetrics.metricsExported)
+	t.Run("with metric/trace drivers configured", func(t *testing.T) {
+		driverTraces := &stubProtocolDriver{}
+		driverMetrics := &stubProtocolDriver{}
+
+		driver := otlp.NewSplitDriver(otlp.WithMetricDriver(driverMetrics), otlp.WithTraceDriver(driverTraces))
+		ctx := context.Background()
+		assert.NoError(t, driver.Start(ctx))
+		assert.Equal(t, 1, driverTraces.started)
+		assert.Equal(t, 1, driverMetrics.started)
+		assert.Equal(t, 0, driverTraces.stopped)
+		assert.Equal(t, 0, driverMetrics.stopped)
+		assert.Equal(t, 0, driverTraces.tracesExported)
+		assert.Equal(t, 0, driverTraces.metricsExported)
+		assert.Equal(t, 0, driverMetrics.tracesExported)
+		assert.Equal(t, 0, driverMetrics.metricsExported)
+
+		recordCount := 5
+		spanCount := 7
+		assert.NoError(t, driver.ExportMetrics(ctx, stubCheckpointSet{recordCount}, metricsdk.StatelessExportKindSelector()))
+		assert.NoError(t, driver.ExportTraces(ctx, stubSpanSnapshot(spanCount)))
+		assert.Len(t, driverTraces.rm, 0)
+		assert.Len(t, driverTraces.rs, spanCount)
+		assert.Len(t, driverMetrics.rm, recordCount)
+		assert.Len(t, driverMetrics.rs, 0)
+		assert.Equal(t, 1, driverTraces.tracesExported)
+		assert.Equal(t, 0, driverTraces.metricsExported)
+		assert.Equal(t, 0, driverMetrics.tracesExported)
+		assert.Equal(t, 1, driverMetrics.metricsExported)
+
+		assert.NoError(t, driver.Stop(ctx))
+		assert.Equal(t, 1, driverTraces.started)
+		assert.Equal(t, 1, driverMetrics.started)
+		assert.Equal(t, 1, driverTraces.stopped)
+		assert.Equal(t, 1, driverMetrics.stopped)
+		assert.Equal(t, 1, driverTraces.tracesExported)
+		assert.Equal(t, 0, driverTraces.metricsExported)
+		assert.Equal(t, 0, driverMetrics.tracesExported)
+		assert.Equal(t, 1, driverMetrics.metricsExported)
+	})
+
+	t.Run("with just metric driver", func(t *testing.T) {
+		driverMetrics := &stubProtocolDriver{}
+
+		driver := otlp.NewSplitDriver(otlp.WithMetricDriver(driverMetrics))
+		ctx := context.Background()
+		assert.NoError(t, driver.Start(ctx))
+
+		assert.Equal(t, 1, driverMetrics.started)
+		assert.Equal(t, 0, driverMetrics.stopped)
+		assert.Equal(t, 0, driverMetrics.tracesExported)
+		assert.Equal(t, 0, driverMetrics.metricsExported)
+
+		recordCount := 5
+		assert.NoError(t, driver.ExportMetrics(ctx, stubCheckpointSet{recordCount}, metricsdk.StatelessExportKindSelector()))
+		assert.NoError(t, driver.ExportTraces(ctx, nil))
```

Comment:
I know this makes no actual difference in the code, but could we change these nils to actual snapshots like the test above?
This is just to make sure future iterations of this can't shortcut nil or 0 len inputs and still pass but fail with real data.

Reply:
It would make sense to just abstract this part of the repeated code into a separate function that can be called and will ensure this uniform call.

Reply:
That makes a lot of sense. I will work on refining the test cases, was trying to get the core change out for review. Do you want me to change this to a draft PR @MrAlias ?

Reply:
@bryan-aguilar if you think the changes needed will be greater than just the two tests here doing in another PR sounds reasonable, but if not including the changes here seems valid.